x/pylons/types/genesis_test.go

```diff
@@ -1,17 +1,146 @@
 package types
 
 import (
+	fmt "fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 )
 
 func TestValidateGenesis(t *testing.T) {
-	genesis := DefaultGenesis()
-	err := genesis.Validate()
-	require.NoError(t, err)
-
-	genesis = NetworkTestGenesis()
-	err = genesis.Validate()
-	require.NoError(t, err)
+	for _, tc := range []struct {
+		desc string
+		req  *GenesisState
+		err  error
+	}{
+		{
+			desc: "Valid 1",
+			req:  DefaultGenesis(),
+		},
+		{
+			desc: "Valid 2",
+			req:  NetworkTestGenesis(),
+		},
+		{
+			desc: "Duplicate index of Redeminfo",
+			req: &GenesisState{
+				RedeemInfoList: []RedeemInfo{
+					{Id: "test"},
+					{Id: "test"},
+				},
+			},
+			err: fmt.Errorf("duplicated index for redeemInfo"),
+		},
+		{
+			desc: "Duplicate index of Redeminqo",
```

Comment:
"redeeminfo" instead of "Redeminqo".

Reply:
Thank you very much! I have fix in commit https://github.com/Pylons-tech/pylons/pull/870/commits/8aa682a4948fce1f64f88c04f253cbb7e3b4e2f1